core/common/src/annotation/TextAnnotation.ts

```diff
@@ -140,6 +174,8 @@ export class TextAnnotation {
       props.anchor = { ...this.anchor };
     }
 
+    props.frame = this.frame;
```

Comment:
You should definitely clone this so that the caller can modify the returned JSON object without accidentally mutating the source object.